Simplenote/NSString+Simplenote.swift

```diff
@@ -45,3 +72,43 @@ extension NSString {
         return UnicodeScalar(character(at: location))
     }
 }
+
+
+// MARK: - Lists Convenience API
+//
+extension NSString {
+
+    /// Returns a new AttributedString instance, by inserting list markers at the beginning of each line
+    ///
+    var insertingListMarkers: NSAttributedString {
+        let output = NSMutableAttributedString()
+        let lines = components(separatedBy: .newlines) as [NSString]
+        let indexOfLastLine = lines.count - 1
+
+        for (index, line) in lines.enumerated() {
+            let leading = line.leadingSpaces
+            let payload = line.substring(from: leading.utf16.count)
+            let attachment = SPTextAttachment(tintColor: .textListColor)
+
+            output.append(string: leading)
+            output.append(attachment: attachment)
+            output.append(string: .space)
+            output.append(string: payload)
+
+            if index != indexOfLastLine {
+                output.append(string: .newline)
+            }
+        }
+
+        return output
+    }
+
+    /// Returns a new AttributedString instance, by removing all of the List Markers (with or without trailing spaces) in the receiver
+    ///
+    var removingListMarkers: NSAttributedString {
+        let output = replacingOccurrences(of: .attachmentString + .space, with: String())
```

Comment:
Should `.attachmentString + .space` be `.richListMarker` ?

Reply:
Nice catch!! Thank you!!